### PR TITLE
Option to pad the last validation input sequence if its smaller than the encoder sequence length for MegatronGPT

### PR DESCRIPTION
**Pull request by anmolgupt**
Option to pad the last validation input sequence if its smaller than the encoder sequence length for MegatronGPT

# What does this PR do ?

If validation_drop_last is set to False, the last input sample from the validation dataset will not be dropped if its smaller than the encode sequence length for MegatronGPT

**Collection**: [Note which collection this PR will affect]
nlp/language_modeling/megatron

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/data/language_modeling/megatron/gpt_dataset.py [hunk 1]
```diff
@@ -345,6 +373,9 @@ def __getitem__(self, idx):
         attention_mask, loss_mask, position_ids = _create_ltor_masks_and_position_ids(
             tokens, self.eos_id, self.reset_position_ids, self.reset_attention_mask, self.eod_mask_loss,
         )
+        loss_mask[labels == -1] = 0.0
```
Comment (MaximumEntropy) [inline nemo/collections/nlp/data/language_modeling/megatron/gpt_dataset.py:376]: You are modifying the loss mask here corresponding to the padded labels, but the attention mask seems unchanged. You can no longer use just a regular `tril` mask here, right?

File: nemo/collections/nlp/data/language_modeling/megatron/gpt_dataset.py [hunk 2]
```diff
@@ -345,6 +373,9 @@ def __getitem__(self, idx):
...
+        tokens[tokens == -1] = 0
```
Comment (MaximumEntropy) [inline nemo/collections/nlp/data/language_modeling/megatron/gpt_dataset.py:377]: This is assuming the padding token id is always 0?

**Review by MaximumEntropy — COMMENTED** [review]
Couple of comments - main concern is about not changing the attention mask corresponding to pad tokens.

**Review by MaximumEntropy — APPROVED** [review]
